File: benchmark/hashmap/hashmap.rb
n = 300_000
keys = 20_011
reps = 25

size_sum = 0
total = 0
rep = 0
while rep < reps
  counts = {}
  i = 0
  while i < n
    key = 'k' + ((i % keys) * 7919 % keys).to_s
    if counts.key?(key)
      counts[key] += 1
    else
      counts[key] = 1
    end
    i += 1
  end

  i = 0
  while i < n
    key = 'k' + ((i % keys) * 7919 % keys).to_s
    total += counts[key]
    i += 1
  end

  size_sum += counts.size
  rep += 1
end

puts size_sum
puts total

File: benchmark/hashmap/hashmap.php
<?php

$n = 300000;
$keys = 20011;
$reps = 25;

$sizeSum = 0;
$total = 0;
for ($rep = 0; $rep < $reps; $rep += 1) {
    $counts = [];
    for ($i = 0; $i < $n; $i += 1) {
        $key = 'k' . (($i % $keys) * 7919 % $keys);
        if (isset($counts[$key])) {
            $counts[$key] += 1;
        } else {
            $counts[$key] = 1;
        }
    }

    for ($i = 0; $i < $n; $i += 1) {
        $key = 'k' . (($i % $keys) * 7919 % $keys);
        $total += $counts[$key];
    }

    $sizeSum += count($counts);
}

echo $sizeSum, "\n";
echo $total, "\n";
